llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp

```diff
@@ -6957,8 +6957,7 @@ void SelectionDAGBuilder::visitIntrinsicCall(const CallInst &I,
     SDValue Result;
     Result = DAG.getNode(
         ISD::FPTRUNC_ROUND, sdl, VT, getValue(I.getArgOperand(0)),
-        DAG.getTargetConstant((int)*RoundMode, sdl,
-                              TLI.getPointerTy(DAG.getDataLayout())));
+        DAG.getTargetConstant((int)*RoundMode, sdl, MVT::i32));
```

Comment:
This is also a separate patch

Reply:
Done!

Reply:
We may still need to change the type to i32, otherwise we will have to change the pattern matching for the type.

Reply:
Also need to be consistent with global-isel path for type of i32. (pattern matching may be confused if not the same)

Reply:
done pre-commit!